release notes/v0.29.0.md

```diff
@@ -0,0 +1,133 @@
+// Download addsvc.proto for https://grpcb.in/, located at:
+// https://raw.githubusercontent.com/moul/pb/master/addsvc/addsvc.proto
+// and put it in the same folder as this script.
+client.load(null, "addsvc.proto")
+
+export default () => {
+    client.connect("grpcb.in:9001", { timeout: "5s" });
+
+    let response = client.invoke("addsvc.Add/Sum", {
+        a: 1,
+        b: 2
+    })
+    console.log(response.message.v) // should print 3
+
+    client.close()
+}
+```
+
+You can find more information and examples how to use k6's new gRPC testing capabilities in [our documentation](https://k6.io/docs/javascript-api/k6-net-grpc).
+
+Huge thanks to @rogchap for adding this feature!
+
+### New options for configuring DNS resolution ([#1612](https://github.com/loadimpact/k6/pull/1612))
+
+You can now control some aspects of how k6 performs DNS resolution! Previously, k6 would have cached DNS responses indefinitely ([#726](https://github.com/loadimpact/k6/issues/726)) and always picked the first resolved IP ([#738](https://github.com/loadimpact/k6/issues/738)) for all connections. This caused issues, especially when load testing services that relied on DNS for load-balancing or auto-scaling.
+
+For technical reasons explored in ([#726](https://github.com/loadimpact/k6/issues/726)), k6 v0.29.0 still doesn't respect the actual TTL value of resolved IPs, that will be fixed in a future k6 version. For now, it simply allows users to specify a global static DNS TTL value and resolution strategy manually. It also has better defaults! Now, by default, the global DNS TTL value is 5 minutes and, if the DNS resolution returned multiple IPs, k6 will pick a random (preferably IPv4) one for each connection.
+
+You can also configure this behavior with the new `--dns` CLI flag, the `K6_DNS` environment variable, or the `dns` script/JSON option. Only 3 DNS resolution options are exposed in this k6 version, `ttl`, `select`, and `policy`.
+
+Possible `ttl` values are :
+- `0`: no caching at all - each request will trigger a new DNS lookup.
+- `inf`: cache any resolved IPs for the duration of the test run _(the old k6 behavior)_.
+- any time duration like `60s`, `5m30s`, `10m`, `2h`, etc.; if no unit is specified (e.g. `ttl=3000`), k6 assumes milliseconds. _The new default value is `5s`._
+
+Possible `select` values are:
+- `first` - always pick the first resolved IP _(the old k6 behavior)_.
+- `random` - pick a random IP for every new connection _(the new default value)_.
+- `roundRobin` - iterate sequentially over the resolved IPs.
+
+Possible `policy` values are:
+- `preferIPv4`: use IPv4 addresses, if available, otherwise fall back to IPv6.
+- `preferIPv6`: use IPv6 addresses, if available, otherwise fall back to IPv4 _(the new default value)_.
+- `onlyIPv4`: only use IPv4 addresses, ignore any IPv6 ones.
+- `onlyIPv6`: only use IPv6 addresses, ignore any IPv4 ones.
+- `any`: no preference, use all addresses _(the old k6 behavior)_.
```

Comment:
Do we need to repeat all these options docs here? It seems like clutter for the release notes. I think just having the intro, example and a link to the docs would be enough.

Reply:
Maybe for most things I'd agree, for example that's how I handled the gRPC section. But considering this is sort of a breaking change, and the release notes are not all that lengthy, I think it's fine.

Reply:
It still feels like too much detail to have these verbatim from the options page, but OK. :)

Reply:
it's not quite verbatim, it's actually longer :sweat_smile: there is some extra explanation on top and stresses on the differences with the old k6 versions (see the many "_(the old k6 behavior)_" and "_(the new default value)_" mentions)

Reply:
I don't think this was the old behaviour as the IPs were sorted by having ipv4 so ... in practice ... the default was preferIPv4

Reply:
> IPs were sorted by having ipv4

I don't think that was the case, can you point me to the code?

Reply:
I might be wrong .. it seems it orders it based on https://tools.ietf.org/html/rfc6724 here https://github.com/golang/go/blob/f2e186b87754d3f84a692876501c923eb58f2ee4/src/net/dnsclient_unix.go#L716 ... this does seem to check (as per the RFC) if there is at all a way to open a [udp connection to each IP](https://github.com/golang/go/blob/f2e186b87754d3f84a692876501c923eb58f2ee4/src/net/addrselect.go#L41-L56)  and then as far as I can understand orders the [IPv6 first](https://github.com/golang/go/blob/f2e186b87754d3f84a692876501c923eb58f2ee4/src/net/addrselect_test.go#L14) ... so maybe you are right ... but I am not certain and this is too big of a rabbit hole, but I still think that maybe we should remove the "this was previously the default" statement. 

Maybe we should test it on a machine with an actual working ipv6 address but with v0.28.0 ?

Reply:
my thinking is that we didn't do any IP address sorting and didn't have any such preference in k6 before 0.29... What the OS or DNS servers or even the Go runtime did is immaterial, for the purposes of k6 it was like black magic and it just got the first IP that was returned. I think it's not unreasonable to claim that `ttl=inf,select=first,policy=any` was the previous DNS behavior, since if you set these options now, you should get exactly the same behavior as k6 v0.28.0, right?